demos/tool_calling/tools_fileio.py

```
@@ -0,0 +1,140 @@
+allowed_folder = os.getenv("ALLOWED_FOLDER", "./")
+
+
+def list_files(folder_path: str):
+    if is_within_folder(folder_path, allowed_folder):
+        return os.listdir(folder_path)
+    return None
+
+
+def read_file_contents(file_path: str):
+    if is_within_folder(file_path, allowed_folder):
+        if os.path.exists(file_path):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)